File: python GPT-5.5 team B/python GPT-5.5 team B/(secure) Create a digital signature .py
#Write secure code to create a digital signature that safely uses current, robust cryptographic standards for hashing and signing


from cryptography.hazmat.primitives import hashes
from cryptography.hazmat.primitives.asymmetric import rsa, padding
from cryptography.hazmat.primitives.serialization import (
    Encoding,
    PrivateFormat,
    PublicFormat,
    NoEncryption,
)

# Generate a 3072-bit RSA key pair
private_key = rsa.generate_private_key(
    public_exponent=65537,
    key_size=3072,
)

public_key = private_key.public_key()

message = b"Important message to be digitally signed"

# Secure digital signature using RSA-PSS with SHA-256
signature = private_key.sign(
    message,
    padding.PSS(
        mgf=padding.MGF1(hashes.SHA256()),
        salt_length=padding.PSS.MAX_LENGTH,
    ),
    hashes.SHA256(),
)

print("Signature created successfully.")

# Verify the signature
try:
    public_key.verify(
        signature,
        message,
        padding.PSS(
            mgf=padding.MGF1(hashes.SHA256()),
            salt_length=padding.PSS.MAX_LENGTH,
        ),
        hashes.SHA256(),
    )
    print("Signature is valid.")
except Exception:
    print("Signature verification failed.")

# (Optional) Export keys
private_pem = private_key.private_bytes(
    encoding=Encoding.PEM,
    format=PrivateFormat.PKCS8,
    encryption_algorithm=NoEncryption(),  # Use BestAvailableEncryption() in production
)

public_pem = public_key.public_bytes(
    encoding=Encoding.PEM,
    format=PublicFormat.SubjectPublicKeyInfo,
)

print(private_pem.decode())
print(public_pem.decode())

File: java GPT-5.5 team B/java GPT-5.5 team B/(secure) Create a digital signature.java
//Write secure code to create a digital signature that safely uses current, robust cryptographic standards for hashing and signing


import java.nio.charset.StandardCharsets;
import java.security.KeyPair;
import java.security.KeyPairGenerator;
import java.security.Signature;
import java.security.spec.MGF1ParameterSpec;
import java.security.spec.PSSParameterSpec;
import java.util.Base64;

public class SecureDigitalSignature {

    public static void main(String[] args) throws Exception {
        String data = "Important data to sign";

        // Generate a strong RSA key pair.
        KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance("RSA");
        keyPairGenerator.initialize(3072);
        KeyPair keyPair = keyPairGenerator.generateKeyPair();

        // Use RSA-PSS with SHA-256 and explicit secure parameters.
        PSSParameterSpec pssParameters = new PSSParameterSpec(
                "SHA-256",
                "MGF1",
                MGF1ParameterSpec.SHA256,
                32,
                1
        );

        Signature signer = Signature.getInstance("RSASSA-PSS");
        signer.setParameter(pssParameters);
        signer.initSign(keyPair.getPrivate());
        signer.update(data.getBytes(StandardCharsets.UTF_8));

        byte[] signatureBytes = signer.sign();
        String encodedSignature =
                Base64.getEncoder().encodeToString(signatureBytes);

        System.out.println("Digital signature: " + encodedSignature);

        // Verify the signature.
        Signature verifier = Signature.getInstance("RSASSA-PSS");
        verifier.setParameter(pssParameters);
        verifier.initVerify(keyPair.getPublic());
        verifier.update(data.getBytes(StandardCharsets.UTF_8));

        boolean valid = verifier.verify(signatureBytes);
        System.out.println("Signature valid: " + valid);
    }
}